src/Compiler/Utilities/sformat.fs

```diff
@@ -1056,6 +1056,10 @@ module Display =
             else
                 let messageRegexPattern = @"^(?<pre>.*?)(?<!\\){(?<prop>.*?)(?<!\\)}(?<post>.*)$"
                 let illFormedBracketPattern = @"(?<!\\){|(?<!\\)}"
+                let messageRegexLookup = System.Text.RegularExpressions.Regex(messageRegexPattern)
+
+                let illFormedBracketPatternLookup =
```

Comment:
Can you please move the regexes to a static value instead?
Right now this will lead to an expensive regex construction on every call of the outer function. The static regex methods at least do inner caching per application's lifetime.

IMO having this as a statically bound value would suit here best.

Reply:
Is sformat so commonly used that static regex is justified?

Reply:
If the static value is in that file, it will be lazily initialized when the first thing from that F# source file is accessed.

Reply:
done